### PR TITLE
Extending dashboard

### PR DESCRIPTION
**Pull request by jwm0**
I want to merge this change because...

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://feature-extending-dashboard-with-apps.api.saleor.rocks/graphql/

### CODE REVIEW

File: src/components/ButtonWithSelect/ButtonWithSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        anchorEl={anchorRef.current}
+        transition
+        disablePortal
+        placement="bottom-end"
+        className={classes.popper}
+      >
+        {({ TransitionProps }) => (
+          <Grow {...TransitionProps}>
+            <Paper className={classes.paper}>
+              <ClickAwayListener onClickAway={handleClose}>
```
Comment (dominik-zeglen) [inline src/components/ButtonWithSelect/ButtonWithSelect.tsx:97]: `@material-ui/core/Menu` doesn't meet requirements?

File: src/products/components/ProductListPage/ProductListPage.tsx [hunk 2]
```diff
@@ -129,6 +135,22 @@ export const ProductListPage: React.FC<ProductListPageProps> = props => {
...
+    label,
+    testId: `extension-${id}`,
+    onSelect: open
+  }));
+
+  const extensionCreateButtonItems = create.map(({ label, id, open }) => ({
+    label,
+    testId: `extension-${id}`,
+    onSelect: open
+  }));
```
Comment (dominik-zeglen) [inline src/products/components/ProductListPage/ProductListPage.tsx:153]: Helper map function?

File: src/products/components/ProductUpdatePage/ProductUpdatePage.tsx [hunk 3]
```diff
@@ -243,6 +249,17 @@ export const ProductUpdatePage: React.FC<ProductUpdatePageProps> = ({
...
+  const { moreActions } = useExtensions(
+    AppExtensionViewEnum.PRODUCT,
+    AppExtensionTypeEnum.DETAILS
+  );
+
+  const extensionMenuItems = moreActions.map(({ label, id, open }) => ({
+    label,
+    testId: `extension-${id}`,
+    onSelect: open
+  }));
```
Comment (dominik-zeglen) [inline src/products/components/ProductUpdatePage/ProductUpdatePage.tsx:261]: Same as above

File: src/apps/components/AppDialog/AppDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  IconButton,
+  Typography
+} from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import React from "react";
+
+import { useStyles } from "./styles";
+
+interface AppDialogProps extends DialogProps {
+  onClose(): void;
```
Comment (dominik-zeglen) [inline src/apps/components/AppDialog/AppDialog.tsx:15]: Let's be consistent with the rest of the project

```suggestion
  onClose: () => void;
```

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by orzechdev — APPROVED** [review]

File: src/apps/components/AppFrame/useAppActions.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        sendResponseStatus(actionId, true);
+        return;
+      }
+      default: {
+        sendResponseStatus(action?.payload?.actionId, false);
+        return;
+      }
+    }
+  };
```
[the remark this answers is not in the document]
Comment (jwm0) [reply src/apps/components/AppFrame/useAppActions.ts:41]: Moving it to another file means also moving in all the outside depedencies (like `navigate` which comes from hook), I see your point but I'd wait before this gets big enough and then try to move it outside.

File: src/components/ButtonWithSelect/ButtonWithSelect.tsx (hunk 1 above)
Comment (jwm0) [reply src/components/ButtonWithSelect/ButtonWithSelect.tsx:97]: Menu doesn't have a placement prop

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**